### PR TITLE
updates in inline post and restart files and bug fix in ugwd

### PR DESCRIPTION
**Pull request by junwang-noaa**
This commit include code changes from following tickets:

#12: add fh00 post control file, 
#25: add capability to write out restart files at specified forecast hours
#26: fix bug in ugwd
#15: Add support for GEFS-Aerosols restart capability
#16: Bug fixes in support of FV3GFS-AQM  

Pull request related to this commit:
https://github.com/NOAA-EMC/GFDL_atmos_cubed_sphere/pull/9
https://github.com/ufs-community/ufs-weather-model/pull/16

### CODE REVIEW

File: atmos_model.F90 [hunk 1]
```diff
@@ -944,17 +945,21 @@ end subroutine update_atmos_model_state
...
   integer :: ierr
 #endif
 
 !-----------------------------------------------------------------------
 !---- termination routine for atmospheric model ----
                                               
-    call atmosphere_end (Atmos % Time, Atmos%grid)
-    call FV3GFS_restart_write (IPD_Data, IPD_Restart, Atm_block, &
-                               IPD_Control, Atmos%domain)
+!    if (mpp_pe() == mpp_root_pe() )print *,'in atmos_model_end,restart_endfcst=',restart_endfcst
```
Comment (DusanJovic-NOAA) [inline atmos_model.F90]: Consider removing this debug print statement

File: atmos_model.F90 [hunk 2]
```diff
@@ -944,17 +945,21 @@ end subroutine update_atmos_model_state
...
+    call atmosphere_end (Atmos % Time, Atmos%grid, restart_endfcst)
+    if(restart_endfcst) then
+      call FV3GFS_restart_write (IPD_Data, IPD_Restart, Atm_block, &
+                                 IPD_Control, Atmos%domain)
+!      if (mpp_pe() == mpp_root_pe() )print *,'in atmos_model_end,write final restart'
```
Comment (DusanJovic-NOAA) [inline atmos_model.F90]: Consider removing this debug print statement

File: atmos_model.F90 (hunk 1 above)
Comment (junwang-noaa) [reply atmos_model.F90]: removed

File: atmos_model.F90 (hunk 2 above)
Comment (junwang-noaa) [reply atmos_model.F90]: removed

**Review by rmontuoro — APPROVED** [review]

**Review by rmontuoro — APPROVED** [review]

**Review by SMoorthi-emc — APPROVED** [review]

**Review by SMoorthi-emc — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by climbfuji** [comment]
Note: this commit contains changes to IPD physics that were not made to the CCPP physics. These missing changes will be applied in https://github.com/NCAR/ccpp-physics/pull/372 and associated PRs listed in there.

**Pull request merged**